runelite-client/src/main/java/net/runelite/client/plugins/boosts/BoostsPlugin.java

```diff
@@ -191,8 +194,54 @@ public void addStatChangeIndicator()
 		}
 	}
 
+	/**
+	 * Calculates the amount of time until boosted stats decay,
+	 * accounting for the effect of preserve prayer.
+	 * Preserve extends the time of boosted stats by 50% while active.
+	 * The length of a boost is split into 4 sections of 15 seconds each.
+	 * If the preserve prayer is active for the entire duration of the final
+	 * section it will "activate" adding an additional 15 second section
+	 * to the boost timing. If again the preserve prayer is active for that
+	 * entire section a second 15 second section will be added.
+	 *
+	 * Preserve is only required to be on for the 4th and 5th sections of the boost timer
+	 * to gain full effect (seconds 45-75).
+	 *
+	 * @return integer value in seconds until next boost change
+	 */
 	public int getChangeTime()
 	{
-		return 60 - (int) Duration.between(lastChange, Instant.now()).getSeconds();
+		int timeSinceChange = timeSinceLastChange();
+
+		if (client.isPrayerActive(Prayer.PRESERVE) && timeSinceChange < 45)
```

Comment:
this block can probably just be merged with next if (e.g timeSinceChange < 45 || preserveBeenActive)